### ISSUE TITLE
 Apm publish: The vue dependency range in the package.json file is invalid: csp

### ISSUE THREAD

**Issue by quintanar401** [atom/apm#431]
When I try to publish a package with this deps:
```
"dependencies": {
    "long": "2.x",
    "vue": "csp"
  }
```
I get
```
The vue dependency range in the package.json file is invalid: csp
```
I guess the direct git-url will not work either. 'Update dependencies' cmd can understand this though. Is it possible to use branch names in deps at all?

**Comment by izuzak** [comment]
@quintanar401 Does npm support supplying only a branch name? I don't see that mentioned here:

https://docs.npmjs.com/files/package.json#dependencies
https://docs.npmjs.com/files/package.json#git-urls-as-dependencies
https://docs.npmjs.com/files/package.json#github-urls

> I guess the direct git-url will not work either.

Just wondering -- did you actually try this or are you really just guessing? Using a full URL should work, but using user/repo#branch probably won't and there's already and open issue for that: https://github.com/atom/apm/issues/302

**Comment by quintanar401** [comment]
Yes, npm supports this - It is mentioned on the versions help page and also I was able to update the package dependencies from atom. I guess it is this one:

tag A specific version tagged and published as tag See npm-tag(1)

Ok, I'll try the direct url on the next publish.

Not sure #302 applies here. Probably I was incorrect saying that it is a branch name - it is a tag. Atom Package Update works with "name": "tag" syntax so probably publish should work too.

**Comment by izuzak** [comment]
> Probably I was incorrect saying that it is a branch name - it is a tag. 

Thanks for clarifying -- yeah, I see that tags are supported. I'll reopen this and rename the title of the issue to be a bit clearer about the problem :+1:
